### DISCUSSION TITLE
Sometimes, when confirm dialoga opens, bottom scheet closes

### DISCUSSION THREAD

**Discussion by vytautas-pranskunas-** [taiga-family/taiga-ui#9572] [Q&A] [answered]
Hi,

i have bottom scheet which triggers confirm modal. Sometimes, not sure when, when modal is opened bottom sheet cloeses.
It closes all the time when i add breakpoints to code. If i stop before opening the modal, bottom scheet always closes when modal is shown. I think this is somehow related to botoom scheet always closing on click outside or loose focus...

when i do ```closable: false``` it does not close. but then i need to implement sepearate button to close it if i do not want to choose anything... Thats not good

**Comment by waterplea** [comment]
Can you make a stackblitz with reproduction? Are you on latest version? We recently refactored bottom sheet a bit, I wasn't able to reproduce your issue on current code.

**Reply by vytautas-pranskunas-** [reply]
Hi, thanks for quick answer

I cannot reproduce it on stackblitz here: https://stackblitz.com/edit/angular-r9sdn8?file=src%2Fapp%2Fconfirm-dialog.service.ts
But i can reproduce it at my repo: https://github.com/vytautas-pranskunas-/closing-bottom-sheet
i added code as in stackblitz however it closes if i add breakpoint before this.dialog.open in app.component.ts

would be greatfull for help

**Reply by waterplea** [reply]
I have tried you repo and the issue you describe does not reproduce for me, I tried desktop and mobile emulation in Chrome several times, sheet is never closed when modal is opened for me. Is there anything specific I need to do? Also you addon-mobile is out of sync there with version 4.10 when the rest is 4.12. After updating it to the latest version the issue is not reproduced for me as well.

On more issue is that repo does not have Taiga UI properly installed, it's missing the plugins:

![image](https://github.com/user-attachments/assets/57767754-7943-4c31-b203-f1746330f52d)

I assume it's just an issue with the repo you've quickly created, otherwise you would definitely notice a lot of things not working properly, but just wanted to highlight that.

**Reply by vytautas-pranskunas-** [reply]
NG_EVENT_PLUGINS are there. You are right regarding fast creation of repo.

As i mentioned what i do i set breakpoint before confirm dialog open, dialog that is opened from the bottomscheet, and then try to open confirm dialog with bottom sheet opened. Then when i hit f5 or play to continue further, upon confirm dialog opening, bottom sheet is closed. I will attach video

p.s. also i was not able to reproduce it after first run off app and thought it is solved, but after i stopped an app and started again issue came back.

https://github.com/user-attachments/assets/1fb45288-9ffc-481b-8992-5543d2665dce

**Reply by vytautas-pranskunas-** [reply]
one more issue. After updating to "@taiga-ui/addon-mobile": "^4.12.0", my bottom sheet stoped closing on outside mouse click
I am not sure is this the same issue but in stackblitz it is acting the same if you click TaigaUI logo section (though is closing if you click right above)

https://stackblitz.com/edit/angular-r9sdn8?file=src%2Fapp%2Fbottom.component.ts,src%2Fapp%2Fapp.template.html,src%2Fapp%2Fapp.component.ts,src%2Fapp%2Fconfirm-dialog.service.ts,src%2Fapp%2Fbottom.component.html&hideExplorer=1

**Reply by waterplea** [reply]
>  After updating to "@taiga-ui/addon-mobile": "^4.12.0", my bottom sheet stoped closing on outside mouse click

Double check that you have `NG_EVENT_PLUGINS`, because that's how I noticed your repo didn't have them after updating to 4.12.0 :) Started working after I added them.

> Then when i hit f5 or play to continue further

Apparently that's how it works, when you click that play it counts as a click to the document. If you click this other play in the DevTools (or F8) it works properly and doesn't close:

![image](https://github.com/user-attachments/assets/0e72f44f-8a2e-4357-96aa-060b541748fe)

**Reply by vytautas-pranskunas-** [reply]
Ok. I will check. What about thing that i canno close bottom sheet now at all?

**Reply by vytautas-pranskunas-** [reply]
I would really appreciat your support on bottom sheet not closing issue.

**Reply by waterplea** [reply]
It closes fine for me on your repository. Like I said, double check `NG_EVENT_PLUGINS`

**Reply by vytautas-pranskunas-** [reply]
this NG_EVENT_PLUGINS i have. It was closing before mobile library update. But it i sno closing in stackblitz: 
https://stackblitz.com/edit/angular-r9sdn8?file=src%2Fapp%2Fbottom.component.ts,src%2Fapp%2Fapp.template.html,src%2Fapp%2Fapp.component.ts,src%2Fapp%2Fconfirm-dialog.service.ts,src%2Fapp%2Fbottom.component.html&hideExplorer=1

Please click on top part of tha page where Taiga UI header is displayed!

https://github.com/user-attachments/assets/369134ed-c341-41ad-a445-40ff2d2d5d2a

https://github.com/user-attachments/assets/65436489-4fcd-4e9d-b392-5f03e67f547b

![Screenshot 2024-10-24 at 16 51 42](https://github.com/user-attachments/assets/780b84b5-cde3-411f-91af-dcab2144a45e)

**Reply by vytautas-pranskunas-** [reply]
I was able to dig to the problem but not sure why it does like this. in my setup i have 

```
const providers = importProvidersFrom(
BrowserAnimationsModule,
...other providers
);

export const appConfig: ApplicationConfig = {
provideAnimations(),
providers,
....
}
```

when i commen BrowserAnimationsModule bottom sheet closes. Hence i cannot reproduce it on stackblitz, but in stackblitz still once clicking on upper area bottom sheet does not close.

have you any idea why BrowserAnimationsModule has an impact?

**Reply by waterplea** [reply, marked as answer]
Yeah, animations should be provided using `provideAnimations()` since that function shipped. Module is outdated and when using `BrowserAnimationsModule` you run into this bug which is hard to trace:
https://github.com/angular/angular/issues/37850

As for StackBlitz, header is outside of tui-root and raised above it with z-index: 1 for some reason. I'll remove that z-index.
